src/lib/unlang/interpret.c

```diff
@@ -578,8 +578,8 @@ unlang_frame_action_t frame_eval(request_t *request, unlang_stack_frame_t *frame
 		 *	called the interpreter.
 		 */
 		case UNLANG_ACTION_YIELD:
-			unlang_frame_perf_yield(frame);
-			yielded_set(frame);
+			unlang_frame_perf_yield(&stack->frame[stack->depth]);
+			yielded_set(&stack->frame[stack->depth]);
```

Comment:
Not sure that's right?  The yielding frame should be the frame at the point frame_eval was called?

Reply:
I found that if I used `unlang_function_push` to push a function onto the stack, return `UNLANG_ACTION_PUSHED_CHILD`, then that function returned `UNLANG_ACTION_YIELD`, `frame` was not pointing at the frame which had just yielded.  It may well be that something else is amis, or my methodology is not quite right.

Reply:
Methodology sounds fine, but I'm not sure why the frame would be incorrect.  I've now added asserts to make sure the stack hasn't grown when yield is returned.